File: 53-maximum-subarray/maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_sum=nums[0]
        cursum=0
        for i in nums:
            if(cursum<0):
                cursum=0
            cursum+=i
            max_sum=max(max_sum,cursum)
        return max_sum

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int cursum=0;
        int maxs=nums[0];
        for (int i=0;i<nums.size();i++){
            if(cursum<0){
                cursum=0;
            }
            cursum+=nums[i];
            if(cursum>maxs){
                maxs=cursum;
            }
            
        }
        return maxs;
    }
};